grunt/config/copy.js

```diff
@@ -1,240 +1,207 @@
-                    src: ['<%= languages %>/*.<%= jsonext %>'],
-                    cwd: '<%= sourcedir %>course/',
-                    dest: '<%= outputdir %>course/'
-                }
-            ]
-        }
-    };
+  var nonServerTasks = {
+    courseAssets: {
+      files: [{
+        expand: true,
+        src: ['<%= languages %>/**/*', '!**/*.<%= jsonext %>'],
+        cwd: '<%= sourcedir %>course/',
+        dest: '<%= outputdir %>course/'
+      }]
+    },
+    courseJson: {
+      files: [{
+        expand: true,
+        src: ['<%= languages %>/*.<%= jsonext %>'],
+        cwd: '<%= sourcedir %>course/',
+        dest: '<%= outputdir %>course/'
+      }]
+    }
+  };
 
-    var mandatoryTasks = {
-        index: {
-            files: [
-                {
-                    expand: true,
-                    src: ['<%= sourcedir %>index.html'],
-                    dest: '<%= outputdir %>',
-                    filter: 'isFile',
-                    flatten: true
-                }
-            ]
+  var mandatoryTasks = {
+    index: {
+      files: [{
+        expand: true,
+        src: ['<%= sourcedir %>index.html'],
+        dest: '<%= outputdir %>',
+        filter: 'isFile',
+        flatten: true
+      }]
+    },
+    coreAssets: {
+      files: [{
+        expand: true,
+        src: ['<%= sourcedir %>core/assets/**'],
+        dest: '<%= outputdir %>assets/',
+        rename: _.partial(collate, "assets")
+      }]
+    },
+    componentAssets: {
+      files: [{
+        expand: true,
+        src: ['<%= sourcedir %>components/**/assets/**'],
+        dest: '<%= outputdir %>assets/',
+        filter: function(filepath) {
+          return grunt.config('helpers').includedFilter(filepath);
         },
-        coreAssets: {
-            files: [
-                {
-                    expand: true,
-                    src: ['<%= sourcedir %>core/assets/**'],
-                    dest: '<%= outputdir %>assets/',
-                    rename: _.partial(collate, "assets")
-                }
-            ]
+        rename: _.partial(collate, "assets")
+      }]
+    },
+    componentFonts: {
+      files: [{
+        expand: true,
+        src: ['<%= sourcedir %>components/**/fonts/**'],
+        dest: '<%= outputdir %>fonts/',
+        filter: function(filepath) {
+          return grunt.config('helpers').includedFilter(filepath);
         },
-        componentAssets: {
-            files: [
-                {
-                    expand: true,
-                    src: ['<%= sourcedir %>components/**/assets/**'],
-                    dest: '<%= outputdir %>assets/',
-                    filter: function(filepath) {
-                        return grunt.config('helpers').includedFilter(filepath);
-                    },
-                    rename: _.partial(collate, "assets")
-                }
-            ]
+        rename: _.partial(collate, "fonts")
+      }]
+    },
+    extensionAssets: {
+      files: [{
```

Comment:
@chetan-hajare - Could you put the opening '{' and associated closing '}' on a new line please?